refactor old implementaions, fix errors, re-run examples

Abdo-Eid opened the pull request:
i wish the owner of this package is still alive

Abdo-Eid committed:
refactor old implementaions, fix errors, re-run examples

janfreyberg commented:
Hi, yes I am still alive =)

Thanks for helping to improve the code. Will leave some comments

janfreyberg commented on src/ssvepy/ssvepyepochs.py:
```diff
@@ -1,35 +1,26 @@
+import sklearn.linear_model
+import inspect
 
+EvokedFrequency = collections.namedtuple('EvokedFrequency',
```
why switch to a named tuple? I think the dataclass is more explicit

Abdo-Eid replied:
Thanks for the feedback! Just to clarify, I intentionally switched to a `NamedTuple` here to keep the `EvokedFrequency` containers lightweight, immutable, and closer to a simple result structure, since these are only holding grouped, static outputs.

To be more explicit about the structure, here's the full breakdown of the `NamedTuple` with data types:

```python
from typing import Optional, Union
import numpy as np

EvokedFrequency = collections.namedtuple('EvokedFrequency', [
    'frequencies',   # Optional[np.ndarray] - Array of frequencies, can be None
    'orders',        # Optional[np.ndarray] - Array of orders (harmonics etc.), can be None
    'power',         # Optional[np.ndarray] - Array of power values, can be None
    'snr',           # Optional[np.ndarray] - Array of SNR values, can be None
    'tfr'            # Optional[np.ndarray] - Array for time-frequency results, can be None
])
```

I agree that a `dataclass` can be more explicit in some situations, but for this particular use case where it's just immutable grouped results with no additional methods, I felt a `NamedTuple` keeps things simple and efficient.

That said, if you'd still prefer I convert this back to a `dataclass` for consistency across the codebase, I'm happy to do so — let me know your preference!

janfreyberg commented on src/ssvepy/ssvepyepochs.py:
```diff
@@ -90,278 +81,249 @@ class Ssvep(mne.Epochs):
-        self.info = deepcopy(epochs.info)
-        self.events = deepcopy(epochs.events)
-
+    def __init__(self, epochs, stimulation_frequency,
```
I believe the code previously was formatted using https://github.com/psf/black

It doesn't need to be that exact formatter,  but keeping arguments on separate lines has a reason (it makes reading change diffs easier if e.g. one argument is removed / added)

janfreyberg replied:
please can you run either the black formatter or something like the ruff formatter over the code and update your PR?

Abdo-Eid replied:
i changed poetry to uv and used ruff

janfreyberg commented on src/ssvepy/ssvepyepochs.py:
```diff
@@ -90,278 +81,249 @@ class Ssvep(mne.Epochs):
-                "frequency": self.frequencies,
-            },
+        # PSD calculation: use MNE's compute_psd if not provided
+        if self.psd is None:
```
Thank you, this is nice.

Could you please update the type hint for the psd argument accordingly?

Abdo-Eid replied:
done

Abdo-Eid commented:
## **Refactor `Ssvep` Class for Modern MNE API and Plotting Flexibility**

### **Summary**

This PR refactors the `Ssvep` class to improve compatibility with the latest MNE API, enhance plotting functionality, and clean up the class structure and documentation.

---

### **Key Changes**

* Replaced legacy PSD computation with `epochs.compute_psd()` for modern, stable MNE support.
* Added proper support for `tmin` and `tmax` in PSD calculation.
* Improved type handling and validation for `stimulation_frequency` (supports int, float, list, np.ndarray).
* Cleaned and standardized docstrings for all public methods and class attributes.
* Refactored `_plot_topo` to **fully implement the `channels` argument**, allowing channel selection for topographical plots.
* Improved robustness in plotting methods to handle NaNs, missing data, and edge cases gracefully.
* Preserved backward compatibility for saved files and `load_ssvep` functionality.
